volume-providers/vpc/provider/wait_for_detach_volume.go

```diff
@@ -28,32 +30,51 @@ func (vpcs *VPCSession) WaitForDetachVolume(volumeAttachmentTemplate provider.Vo
 		return err
 	}
 
-	err = vpcs.APIRetry.FlexyRetry(vpcs.Logger, func() (interface{}, error) {
+	err = vpcs.APIRetry.FlexyRetry(vpcs.Logger, func() (error, bool) {
 		currentVolAttachment, err := vpcs.GetVolumeAttachment(volumeAttachmentTemplate)
-		return currentVolAttachment, err
-	}, func(intf interface{}, err *models.Error) bool {
-		// Skip API retry logic, if there is any error keep retry as per configuration
+		vpcs.Logger.Info("Info", zap.Reflect("VolAtt", currentVolAttachment))
+		// In case of error we should not retry as there are two conditions for error
+		// 1- some issues at endpoint side --> Which is already covered in vpcs.GetVolumeAttachment
+		// 2- Attachment not found --> in this case we should not re-try as it has been deleted
 		if err != nil {
-			// stop re-try, as attchment find failed, because volume is already detached
-			if err.Errors[0].Code == userError.VolumeAttachFindFailed {
-				return true
-			}
-			// keep re-try for all other errors
-			return false
+			return err, true
```

Comment:
I would put below condition here 
return err , `err != nil && errMsg.Code != userError.VolumeAttachFindFailed `- Retry only if  error is not `VolumeAttachFindFailed`

Reply:
its same

Reply:
Its not same , we should retry if the error other than `VolumeAttachFindFailed`. This is not happening with this logic

Reply:
not really required as I explained you on call